Application/EdFi.Ods.AdminApi/Features/OdsInstances/DeleteOdsInstance.cs

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using EdFi.Admin.DataAccess.Models;
+using EdFi.Ods.AdminApi.Infrastructure;
+using EdFi.Ods.AdminApi.Infrastructure.Database.Commands;
+using EdFi.Ods.AdminApi.Infrastructure.Database.Queries;
+using EdFi.Ods.AdminApi.Infrastructure.Extensions;
+using FluentValidation;
+
+namespace EdFi.Ods.AdminApi.Features.OdsInstances;
+
+public class DeleteOdsInstance : IFeature
+{
+    public void MapEndpoints(IEndpointRouteBuilder endpoints)
+    {
+        AdminApiEndpointBuilder.MapDelete(endpoints, "/odsInstances/{id}", Handle)
+            .WithDefaultDescription()
+            .WithRouteOptions(b => b.WithResponseCode(200, FeatureConstants.DeletedSuccessResponseDescription))
+            .BuildForVersions(AdminApiVersions.V2);
+    }
+
+    internal async Task<IResult> Handle(IDeleteOdsInstanceCommand deleteOdsInstanceCommand, Validator validator, int id)
+    {
+        var request = new Request() { Id = id };
+        await validator.GuardAsync(request);
+        deleteOdsInstanceCommand.Execute(request.Id);
+        return await Task.FromResult(Results.Ok("Ods Instance".ToJsonObjectResponseDeleted()));
+    }
+
+    public class Validator : AbstractValidator<Request>
+    {
+        private readonly IGetOdsInstanceQuery _getOdsInstanceQuery;
+        private readonly IGetApplicationsByOdsInstanceIdQuery _getApplicationByOdsInstanceIdQuery;
+        private OdsInstance? OdsInstanceEntity = null;
+        public Validator(IGetOdsInstanceQuery getOdsInstanceQuery, IGetApplicationsByOdsInstanceIdQuery getApplicationByOdsInstanceIdQuery)
+        {
+            _getOdsInstanceQuery = getOdsInstanceQuery;
+            _getApplicationByOdsInstanceIdQuery = getApplicationByOdsInstanceIdQuery;
+
+            RuleFor(m => m.Id)
+                .Must(NotHaveRelationships)
+                .WithMessage(FeatureConstants.OdsInstanceCantBeRemovedMessage)
+                .When(Exist);
+        }
+
+        private bool Exist(Request request)
+        {
+            OdsInstanceEntity = _getOdsInstanceQuery.Execute(request.Id) ?? throw new NotFoundException<int>("odsInstance", request.Id);

```

Comment:
@jagudelo-gap 
Can we move the NotFound exception to GetOdsInstanceQuery.? So that no need to check that on multiple places.

Reply:
Done